raiden_libs/network/matrix/client.py

```diff
@@ -156,6 +160,10 @@ def listen_forever(
                     _bad_sync_timeout = min(_bad_sync_timeout * 2, self.bad_sync_timeout_limit)
                 else:
                     raise
+            except MatrixHttpLibError:
```

Comment:
Why are these errors happening?

Reply:
These errors happen mainly when [transport fixture](https://github.com/raiden-network/raiden/blob/077ba5445dae152333c63f062287d42ff8d396cf/raiden/tests/integration/fixtures/transport.py#L58) stops the server because of some other exception that happened in the tests, but then a ton of greenlets which weren't cleared yet may fail with `Connection aborted` error, as `synapse` was terminated, spamming transport errors in the logs and shadowing the actual exception.

Reply:
Why is this class being patched because of tests failures? Why don't you just [cleanup the greenlets](https://github.com/raiden-network/raiden/blob/077ba5445dae152333c63f062287d42ff8d396cf/raiden/tests/integration/fixtures/raiden_network.py#L194)? Is the transport fixture being used without the network fixtures?

Reply:
@hackaugusto In my tests, somehow the transport fixture killed the `synapse` server before the cleanup tasks had the opportunity to clean these greenlets. It's being done here because it's an invariant from the `MatrixTransport` perspective that the server will be available (or at least an exception related to it not being raised) during the lifetime of the Transport. Notice this PR improves a behavior which is already intended in this class.

Reply:
> In my tests, somehow the transport fixture killed the synapse server before the cleanup tasks had the opportunity to clean these greenlets. 

Okay, so the problem is the order in which things are happening, first the synapse server is killed, then the greenlets are not clean, so they try to make a request, and then the greenlets are killed, which is too late because they have failed already. Correct?

Reply:
Yes.

Reply:
The above cannot be true, that is pytest's behavior. [example](https://gist.github.com/hackaugusto/d66fc1b4206f785327ab3d8a03979b22)

Reply:
> But again, this Connection aborted error shouldn't be fatal, these long polling http requests can raise it for a number of reasons, it should be just retried, and when desired the transport should bestoped

Agreed, but what is the difference among the exceptions `MatrixHttpLibError` and `MatrixRequestError`?

Reply:
`MatrixHttpLibError` are connection/socket related, and no proper HTTP response was received. `MatrixRequestError` is when `synapse` answers with an actual HTTP response but which is an error on itself, like `403`. `MatrixRequestError` with `code >= 500` means something in the server or frontend connection (unhandled server exception, load balancers, etc) didn't go right, and may be handled like `MatrixHttpLibError`, and retried expecting to work next time, or timeout.

Reply:
Okay, so handling this exceptions seems right, but I don't agree on the diagnosis of why they are happening in the first place. The explanation that synapse is killed before the requesting greenlets didn't make sense to me.

Reply:
Well, be my guest to debug this thing by yourself then and providing a better explanation. This PR fixed the tests in my case, while forcing a random exception on unrelated greenlet led to Matrix log spam I mentioned, and this PR shows the actual exception on.